jupyterhub/proxy.py

```diff
@@ -300,7 +301,7 @@ def check_routes(self, user_dict, service_dict, routes=None):
             user = user_dict[orm_user]
             if user.running:
                 good_routes.add(user.proxy_spec)
-                if user.name not in user_routes:
+                if user.name not in user_routes and not user.proxy_pending:
```

Comment:
At the moment, this `not user_proxy_pending` will always be true because `user.running` is contingent on `user.proxy_pending` being False.

We need to make sure that nether `add_user` nor `delete_user` is called if `user.proxy_pending`

I think this should fix the mishandled case:

```python
if user.running:
    # unchanged
elif user.proxy_pending:
    good_routes.add(user.proxy_spec)
```